controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -212,12 +213,15 @@ func (r *BareMetalHostReconciler) Reconcile(ctx context.Context, request ctrl.Re
 	}
 
 	ready, err := prov.TryInit()
-	if err != nil {
-		return ctrl.Result{}, errors.Wrap(err, "failed to check services availability")
-	}
-	if !ready {
+	if err != nil || !ready {
```

Comment:
Log-wise, how is it different from just return an error if err != nil?

Reply:
in some of the implementation of TryInit(), it was returning ready state as bool 'false' and at the same time err was nil. Also this is the standard followed in the code base